build-artifacts/project-template-gradle/src/main/java/com/tns/ErrorReport.java

```diff
@@ -62,66 +104,81 @@ static boolean startActivity(final Context context, String errorMessage)
 		return true;
 	}
 
-	static void killProcess(Context context)
-	{
+	static void killProcess(Context context) {
 		// finish current activity and all below it first
-		if (context instanceof Activity)
-		{
+		if (context instanceof Activity) {
 			((Activity) context).finishAffinity();
 		}
 
 		// kill process
 		android.os.Process.killProcess(android.os.Process.myPid());
 	}
 
-	static void startPendingErrorActivity(Context context, Intent intent) throws CanceledException
-	{
+	static void startPendingErrorActivity(Context context, Intent intent) throws CanceledException {
 		PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_CANCEL_CURRENT);
 
 		pendingIntent.send(context, 0, intent);
 	}
 
-	static String getErrorMessage(Throwable ex)
-	{
+	static String getErrorMessage(Throwable ex) {
 		String content;
 		PrintStream ps = null;
 
-		try
-		{
+		try {
 			ByteArrayOutputStream baos = new ByteArrayOutputStream();
 			ps = new PrintStream(baos);
 			ex.printStackTrace(ps);
 
-			try
-			{
+			try {
 				content = baos.toString("US-ASCII");
-			}
-			catch (UnsupportedEncodingException e)
-			{
+			} catch (UnsupportedEncodingException e) {
 				content = e.getMessage();
 			}
-		}
-		finally
-		{
+		} finally {
 			if (ps != null)
 				ps.close();
 		}
 
 		return content;
 	}
 
-	static Intent getIntent(Context context)
-	{
-		Class<?> errorActivityClass = ErrorReportActivity.class;
+	/*
+	* Gets the process Id of the running app and filters all
+	* output that doesn't belong to that process
+	* */
+	static String getLogcat() {
+		String content;
+		String pId = Integer.toString(android.os.Process.myPid());
```

Comment:
it's better to take the whole log or to make the filter better, because the PID is different before the error activity is shown and hence we're missing information.